### PR TITLE
Fixes #12007: Enforce with a test the behaviour of generic_variable_definition

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/12007

### CODE REVIEW

**Comment by ncharles** [comment]
are you sure you want to change the permissions of hooks ?

File: rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala [hunk 1]
```diff
@@ -701,8 +759,40 @@ class WriteSystemTechniquesTest extends TechniquesTest{
...
+          , Set(root.id, cfeNode.id)
+          , Map(root.id -> rnc, cfeNode.id -> cfeNC)
+          , Map(root.id -> NodeConfigId("root-cfg-id"), cfeNode.id -> NodeConfigId("cfe-node-cfg-id"))
+          , Map(), globalPolicyMode, DateTime.now
+      )
+
+      (writen mustFull) and
+      compareWith(rootPath.getParentFile/cfeNode.id.value, "node-gen-var-def-override",
+//           """.*rudder_common_report\("ntpConfiguration".*@@.*"""  //clock reports
+//        :: """.*add:default:==:.*"""                               //rpm reports
```
Comment (ncharles) [inline rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala]: you'll want to remove this comment

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**